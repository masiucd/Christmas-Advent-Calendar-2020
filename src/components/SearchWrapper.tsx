import * as React from 'react';
import Search from './Search';

interface Props {}

const SearchWrapper: React.FC<Props> = () => {
  const [searchText, setSearchText] = React.useState<string>('');

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
    setSearchText(e.target.value);
  };
  return (
    <div>
      <Search value={searchText} onChange={handleChange} onPlaceholder='search'>
        Search:
      </Search>

      <h3>Searching for {searchText ? searchText : '...'}</h3>
    </div>
  );
};
export default SearchWrapper;
