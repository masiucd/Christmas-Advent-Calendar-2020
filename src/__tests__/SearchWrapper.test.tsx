import React from 'react';
import { render, screen } from '@testing-library/react';
import SearchWrapper from '../components/SearchWrapper';

describe('Search Wrapper', () => {
  test('<SearchWrapper />', () => {
    const { container } = render(<SearchWrapper />);
    expect(screen.getByText('Search:')).toBeInTheDocument();

    expect(screen.getByRole('textbox')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('search')).toBeInTheDocument();
    expect(screen.queryByPlaceholderText('apa')).not.toBeInTheDocument();

    expect(screen.queryByText(/I am the best/)).toBeNull();
    expect(container.firstChild).toMatchSnapshot();
  });
});
