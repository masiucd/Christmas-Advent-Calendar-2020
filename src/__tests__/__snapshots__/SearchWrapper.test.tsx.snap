// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Wrapper <SearchWrapper /> 1`] = `
<div>
  <div>
    <label
      for="search"
    >
      Search:
    </label>
    <input
      id="search"
      placeholder="search"
      type="text"
      value=""
    />
  </div>
  <h3>
    Searching for 
    ...
  </h3>
</div>
`;

exports[`Search Wrapper <SearchWrapper/> 1`] = `
<div>
  <div>
    <label
      for="search"
    >
      Search:
    </label>
    <input
      id="search"
      placeholder="search"
      type="text"
      value=""
    />
  </div>
  <h3>
    Searching for 
    ...
  </h3>
</div>
`;
